### PR TITLE
Header + MnemonicWithPassphrase JSON codable | fix SargonObjectCodable

### PR DESCRIPTION
**Pull request by CyonAlexRDX**
# Rust
UF export JSON data roundtrip for 
- Header
- MnemonicWithPassphrase

# Swift
Fix bug with JSON in Swift, `0` got translated into `false` by NSDictionaryConversion. New solution is bug free and shorter and sweeter, thx to SwiftyJSON

Swift Codable support for:
Header
MnemoncWithPassphrase

### CODE REVIEW

File: apple/Sources/Sargon/Util/SargonObjectCodable.swift [hunk 1]
```diff
@@ -8,16 +8,14 @@ public protocol SargonObjectCodable: Codable {
 
 extension SargonObjectCodable {
 	public func encode(to encoder: any Encoder) throws {
-		let dict = try JSONSerialization.jsonObject(with: self.jsonData(), options: []) as! NSDictionary
```
Comment (CyonAlexRDX) [inline apple/Sources/Sargon/Util/SargonObjectCodable.swift:11]: This line going via NSDictionary caused `0` => `false` bug. 

Might have been fixed passing some options... but the overall solution with SwiftyJSON is better

**Review by micbakos-rdx — APPROVED** [review]

File: src/core/utils/mod.rs [hunk 2]
```diff
@@ -5,3 +5,21 @@ mod string_utils;
 pub use factory::*;
 pub use logged_panic::*;
 pub use string_utils::*;
+
+pub fn type_name<T>() -> String {
```
Comment (CyonAlexRDX) [inline src/core/utils/mod.rs:9]: use this to get name of a type

File: src/profile/v100/header/header.rs [hunk 3]
```diff
@@ -129,6 +119,33 @@ impl HasSampleValues for Header {
     }
 }
 
+impl Header {
+    /// Creates a new `Header` from json in the form of Vec<u8>.
+    /// This is a temporarily exported method that allows wallet clients to
+    /// integrate Profile in steps.
+    ///
+    /// Should be replaced later with `Wallet`
+    pub fn new_from_json_bytes(json: impl AsRef<[u8]>) -> Result<Self> {
```
Comment (CyonAlexRDX) [inline src/profile/v100/header/header.rs:128]: FIXME NOW: remove

File: src/profile/v100/header/header_uniffi_fn.rs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+/// Instantiates a new `Header` with creating and last used on `DeviceInfo` with
+/// "Unknown device" as description, and empty content hint
+#[uniffi::export]
+pub fn new_header_with_creating_device(creating_device: DeviceInfo) -> Header {
+    Header::new(creating_device)
+}
+
+#[uniffi::export]
+pub fn new_header_from_json_bytes(json_bytes: BagOfBytes) -> Result<Header> {
```
Comment (CyonAlexRDX) [inline src/profile/v100/header/header_uniffi_fn.rs:21]: FIXME NOW: Remove, use macro `json_data_convertible`

**Pull request merged**